packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/LinksFieldInput.tsx

```diff
@@ -1,93 +1,134 @@
-    onClickOutside?.(() =>
-      persistLinksField({
-        primaryLinkUrl: url,
-        primaryLinkLabel: '',
-      }),
-    );
-  };
+  const links = useMemo(
+    () =>
+      [
+        fieldValue.primaryLinkUrl
+          ? {
+              url: fieldValue.primaryLinkUrl,
+              label: fieldValue.primaryLinkLabel,
+            }
+          : null,
+        ...(fieldValue.secondaryLinks ?? []),
+      ].filter(isDefined),
+    [
+      fieldValue.primaryLinkLabel,
+      fieldValue.primaryLinkUrl,
+      fieldValue.secondaryLinks,
+    ],
+  );
 
-  const handleTab = (url: string) => {
-    onTab?.(() =>
-      persistLinksField({
-        primaryLinkUrl: url,
-        primaryLinkLabel: '',
-      }),
-    );
-  };
+  useListenClickOutside({
+    refs: [containerRef],
+    callback: (event) => {
+      event.stopImmediatePropagation();
+
+      const isTargetInput =
+        event.target instanceof HTMLInputElement &&
+        event.target.tagName === 'INPUT';
+
+      if (!isTargetInput) {
+        onCancel?.();
+      }
+    },
+  });
+
+  const [isInputDisplayed, setIsInputDisplayed] = useState(false);
+  const [inputValue, setInputValue] = useState('');
 
-  const handleShiftTab = (url: string) => {
-    onShiftTab?.(() =>
+  useScopedHotkeys(Key.Escape, onCancel ?? (() => {}), hotkeyScope);
+
+  const handleSubmit = () => {
+    if (!inputValue) return;
+
+    setIsInputDisplayed(false);
+    setInputValue('');
+
+    if (!links.length) {
       persistLinksField({
-        primaryLinkUrl: url,
+        primaryLinkUrl: inputValue,
         primaryLinkLabel: '',
-      }),
-    );
-  };
+        secondaryLinks: [],
+      });
+
+      return;
+    }
 
-  const handleChange = (url: string) => {
-    setDraftValue({
-      primaryLinkUrl: url,
-      primaryLinkLabel: '',
+    persistLinksField({
+      ...fieldValue,
+      secondaryLinks: [
+        ...(fieldValue.secondaryLinks ?? []),
+        { label: '', url: inputValue },
+      ],
     });
   };
 
   return (
-    <FieldInputOverlay>
-      <TextInput
-        value={draftValue?.primaryLinkUrl ?? ''}
-        autoFocus
-        placeholder="Links"
-        hotkeyScope={hotkeyScope}
-        onClickOutside={handleClickOutside}
-        onEnter={handleEnter}
-        onEscape={handleEscape}
-        onTab={handleTab}
-        onShiftTab={handleShiftTab}
-        onChange={handleChange}
-      />
-    </FieldInputOverlay>
+    <StyledDropdownMenu ref={containerRef} width={200}>
+      {!!links.length && (
+        <>
+          <DropdownMenuItemsContainer>
+            {links.map(({ label, url }, index) => (
+              <MenuItem
+                key={index}
+                text={<LinkDisplay value={{ label, url }} />}
+              />
+            ))}
+          </DropdownMenuItemsContainer>
+          <DropdownMenuSeparator />
+        </>
+      )}
+      {isInputDisplayed ? (
+        <DropdownMenuInput
```

Comment:
Why are we breaking the explicit pattern that was here previously and that makes all our input consistent ? It makes it hard to understand why we're passing inline callbacks instead of explicit handlers.

Reply:
What is the pattern that should be followed? I followed `RelationFieldInput` as an example, which only accepts `onCancel` and `onSubmit` as props, and is a dropdown input as well. For instance, `SingleEntitySelect` defines its own `useListenClickOutside` and does not use `onClickOutside` from `FieldInput`. I am not sure how to leverage the handlers passed by `FieldInput` when the input is a Dropdown, as the Dropdown interface is different from other internal inputs?

Reply:
I checked with Charles, the UX decision is to not use onCancel even if click outside or escape. 

So here we want to have every event using this onSubmit.

Thus draft value is useless here, but we should keep the onSubmit FieldInputEvent like the other Fields to be consistent.